Runtime/Interaction/SnapZone.cs

```diff
@@ -317,24 +337,45 @@ public override void ProcessInteractor(XRInteractionUpdateOrder.UpdatePhase upda
             {
                 base.ProcessInteractor(updatePhase);
             }
-
+            
             if (socketActive)
             {
-                if (m_HoverTargets.Count == 0 && ShowHighlightObject)
+                hoverTargets.RemoveAll(target => target == null || target.enabled == false);
+                
+                CheckForReleasedHoverTargets();
+                
+                ShowHighlight();
+            }
+        }
+
+        private void CheckForReleasedHoverTargets()
+        {
+            hoverTargets.Any(target =>
```

Comment:
My previous suggestion on improving this LINQ still apply.

Reply:
Well it's suggestion, I dont see how this is actually better. But using linq here in anyway was actually wrong decision here so i removed the usage.